src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -557,9 +559,18 @@ - (MTRDeviceController * _Nullable)_startDeviceController:(MTRDeviceController *
         return nil;
     }
 
-    if ([_controllers count] == 0) {
-        dispatch_sync(_chipWorkQueue, ^{
-            self->_operationalBrowser = new MTROperationalBrowser(self, self->_chipWorkQueue);
+    // Listen for changes to suspended state on this controller, so we can
+    // track whether we have any unsuspended controllers.
+    [controller addDeviceControllerDelegate:self queue:_chipWorkQueue];
```

Comment:
I guess this works, but it does feel slightly odd to have the factory add itself as a generic delegate here to implement the operational browse behavior. I think it would be maybe more obvious if the MTRDeviceController itself was simply directly responsible for calling counted Start and Stop methods on the browser as appropriate when it starts/stop/suspends/resumes.

Reply:
I considered having the controller notify the controller factory directly via some sort of out-of-band API, but just reusing the existing delegate API seemed simpler.

But yes, we could expose the browser to controllers and have them call methods on it directly.  Maybe that will be simpler; I'll code it up and see how it looks.

Reply:
exposing the browser to controllers makes sense to me